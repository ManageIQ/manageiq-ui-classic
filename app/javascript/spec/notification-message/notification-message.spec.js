import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import NotificationMessage from '../../components/notification-message';

describe('NotificationMessage component', () => {
  it('should render error message', () => {
    const wrapper = mount(<NotificationMessage type="error" message={__('Error message goes here')} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.alert-danger')).toHaveLength(1);
    expect(wrapper.find('.pficon-error-circle-o')).toHaveLength(1);
  });

  it('should render information message', () => {
    const wrapper = mount(<NotificationMessage type="info" message={__('Information message goes here')} />);
    expect(wrapper.find('.alert-info')).toHaveLength(1);
    expect(wrapper.find('.pficon-info')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render success message', () => {
    const wrapper = mount(<NotificationMessage type="success" message={__('Success message goes here')} />);
    expect(wrapper.find('.alert-success')).toHaveLength(1);
    expect(wrapper.find('.pficon-ok')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render warning message', () => {
    const wrapper = mount(<NotificationMessage type="warning" message={__('Warning message goes here')} />);
    expect(wrapper.find('.alert-warning')).toHaveLength(1);
    expect(wrapper.find('.pficon-warning-triangle-o')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('invalid type should not render message', () => {
    const wrapper = mount(<NotificationMessage type="unknown" message={__('No message will be displayed')} />);
    expect(wrapper.find('.miq-notification-message-container')).toHaveLength(1);
    expect(wrapper.find('.fa-question-circle')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('blank message should not render message', () => {
    const wrapper = mount(<NotificationMessage type="error" message="" />);
    expect(wrapper.find('.miq-notification-message-container')).toHaveLength(0);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('unknown error type message should be rendered with blank type', () => {
    const wrapper = mount(<NotificationMessage type="" message={__('Message goes here')} />);
    expect(wrapper.find('.miq-notification-message-container')).toHaveLength(1);
    expect(wrapper.find('.fa-question-circle')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('unknown error type message should be rendered with no type', () => {
    const wrapper = mount(<NotificationMessage message={__('Message goes here')} />);
    expect(wrapper.find('.miq-notification-message-container')).toHaveLength(1);
    expect(wrapper.find('.fa-question-circle')).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
