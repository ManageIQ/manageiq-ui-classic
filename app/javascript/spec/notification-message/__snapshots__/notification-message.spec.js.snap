// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationMessage component blank message should not render message 1`] = `
<NotificationMessage
  message=""
  type="error"
/>
`;

exports[`NotificationMessage component invalid type should not render message 1`] = `
<NotificationMessage
  message="No message will be displayed"
  type="unknown"
>
  <div
    className="miq-notification-message-container alert alert-info"
  >
    <span
      className="pficon fa fa-regular fa-question-circle"
    />
    <strong>
      No message will be displayed
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component should render error message 1`] = `
<NotificationMessage
  message="Error message goes here"
  type="error"
>
  <div
    className="miq-notification-message-container alert alert-danger"
  >
    <span
      className="pficon pficon-error-circle-o"
    />
    <strong>
      Error message goes here
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component should render information message 1`] = `
<NotificationMessage
  message="Information message goes here"
  type="info"
>
  <div
    className="miq-notification-message-container alert alert-info"
  >
    <span
      className="pficon pficon-info"
    />
    <strong>
      Information message goes here
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component should render success message 1`] = `
<NotificationMessage
  message="Success message goes here"
  type="success"
>
  <div
    className="miq-notification-message-container alert alert-success"
  >
    <span
      className="pficon pficon-ok"
    />
    <strong>
      Success message goes here
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component should render warning message 1`] = `
<NotificationMessage
  message="Warning message goes here"
  type="warning"
>
  <div
    className="miq-notification-message-container alert alert-warning"
  >
    <span
      className="pficon pficon-warning-triangle-o"
    />
    <strong>
      Warning message goes here
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component unknown error type message should be rendered with blank type 1`] = `
<NotificationMessage
  message="Message goes here"
  type=""
>
  <div
    className="miq-notification-message-container alert alert-info"
  >
    <span
      className="pficon fa fa-regular fa-question-circle"
    />
    <strong>
      Message goes here
    </strong>
  </div>
</NotificationMessage>
`;

exports[`NotificationMessage component unknown error type message should be rendered with no type 1`] = `
<NotificationMessage
  message="Message goes here"
>
  <div
    className="miq-notification-message-container alert alert-info"
  >
    <span
      className="pficon fa fa-regular fa-question-circle"
    />
    <strong>
      Message goes here
    </strong>
  </div>
</NotificationMessage>
`;
